UPEI-Android/group-project-mindjogg
# The MVC design pattern set up for server

## Source 

Closes #30 

## Description 

### Summary of PR 

The PR will refactor some code from the server.js and make it as a module and implements the MVC design pattern (excluding View).

### Purpose 

The main goal of the PR is break the server into multiple mini app so that it is easy to maintain.  

## Review and Testing Steps

1. Pull the new branch 
2. Read the code (It is very important to understand why I wrote code in this way. Please ask question if you have any confusion or concerns because we should write code in the same structure in future. I have also added some TODO in the code for who will develop the mini app further)
3. Run the server and see if it works
4. Test the user registration routes and user login routes (can be found in routes folder)
5. Get the user list in the get user list routes 

If everything works then the code are fine otherwise let me know.

### Notes
The PR will not contain server configuration or database connection configuration files.

### Resources 
Want to learn about MVC design pattern in NODE.JS app?  Check out the video : https://www.youtube.com/watch?v=zW_tZR0Ir3Q
